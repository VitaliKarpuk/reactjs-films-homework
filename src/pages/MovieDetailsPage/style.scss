.movie-details {
  max-width: 100%;
  height: 1000px;
  background-image: linear-gradient(
      to bottom,
      rgba(37, 37, 37, 0.808) 0%,
      rgba(70, 68, 68, 0) 20%
    ),
    linear-gradient(to top, rgba(37, 37, 37, 0.808) 0%, rgba(70, 68, 68, 0) 40%),
    url(../../images/header-background.png);
  background-size: 100% 100%;
  background-repeat: no-repeat;
  padding: 50px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  .header {
    display: flex;
    justify-content: space-between;

    h1 {
      font-size: 50px;
      letter-spacing: 2px;
      line-height: 64px;
      color: #ffffff;
      font-weight: 800;
    }

    .header__search {
      display: flex;
      align-items: center;
    }

    input {
      width: 492px;
      height: 25px;
      padding: 20px;
      background-color: transparent;
      border: 3.756702169249623px solid #ffffff;
      font-size: 28px;
      letter-spacing: 1px;
      line-height: 44px;
      color: #ffffff;
      text-align: left;
      border-radius: 7%;
      -webkit-border-radius: 7%;
      -moz-border-radius: 7%;
      -ms-border-radius: 7%;
      -o-border-radius: 7%;

      &::placeholder {
        font-size: 28px;
        letter-spacing: 1px;
        line-height: 44px;
        color: #ffffff;
        font-family: "Helvetica Neue";
        text-align: left;
      }
    }

    span {
      position: relative;
      left: -35px;
      display: block;
      width: 28px;
      height: 28px;
      background-image: url(../../images/Search-Icon.png);
      background-size: 100% 100%;
      background-repeat: no-repeat;
    }
  }
}

.movie-details__content {
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  color: #ffffff;

  .movie-details__title {
    h2 {
      font-size: 60px;
      letter-spacing: 2px;
      line-height: 44px;
      color: #ffffff;
      font-weight: bold;
      margin-bottom: 20px;
    }

    .movie-details__genre {
      list-style: none;
      display: flex;
      font-size: 24px;
      letter-spacing: 1px;
      line-height: 44px;
      color: #ffffff;
      padding: 0;

      li {
        margin-right: 15px;
      }
    }
  }

  .movie-details__buttons {
    width: 600px;
    display: flex;
    flex-wrap: wrap;
  }
}
