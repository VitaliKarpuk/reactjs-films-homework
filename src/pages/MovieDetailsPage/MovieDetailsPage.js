import React from 'react';
import Stars from '../../components/Stars/Stars';
import Btn from '../../components/Btn/Btn';
import ModalInfo from '../../components/ModalInfo/ModalInfo';

import './style.scss';


const MovieDetailsPage = () => {
  return (
    <div className="movie-details">
      <header className='header'>
        <h1>FILMS</h1>
        <div className="header__search">
          <input type="text" placeholder="the jungle book" />
          <span className='header__search-icon'></span>
        </div>
      </header>
      <div className="movie-details__content">
        <div className="movie-details__title">
          <h2>THE JUNGLE BOOK</h2>
          <ul className="movie-details__genre">
            <li>Adventure</li>
            <li>Drama</li>
            <li>Family</li>
            <li>Fantasy</li>
            <li>1h 46m</li>
          </ul>
          <Stars />
        </div>
        <div className="movie-details__buttons">
          <ModalInfo />
          <Btn title='Watch Now' className="btn__watch"/>
          <Btn title='View Info' className="btn__info"/>
        </div>

      </div>
    </div>
  )
}

export default MovieDetailsPage;