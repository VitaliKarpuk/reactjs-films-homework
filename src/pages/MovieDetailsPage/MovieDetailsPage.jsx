import React from 'react';
import Stars from '../../components/Stars/Stars';
import ModalInfo from '../../components/ModalInfo/ModalInfo';
import SearchInput from '../../components/SearchInput/SearchInput';
import Button from '../../components/Button/Button';

import './style.scss';

const MovieDetailsPage = () => {
  return (
    <div className="movie-details">
      <header className='header'>
        <h1>FILMS</h1>
        <SearchInput />
      </header>
      <div className="movie-details__content">
        <div className="movie-details__title">
          <h2>THE JUNGLE BOOK</h2>
          <ul className="movie-details__genre">
            <li>Adventure</li>
            <li>Drama</li>
            <li>Family</li>
            <li>Fantasy</li>
            <li>1h 46m</li>
          </ul>
          <Stars />
        </div>
        <div className="movie-details__buttons">
          <ModalInfo />
          <Button className={'btn__watch'} title={'Watch Now'} />
          <Button className={'btn__info'} title={'View Info'} />
        </div>
      </div>
    </div>
  )
}

export default MovieDetailsPage;