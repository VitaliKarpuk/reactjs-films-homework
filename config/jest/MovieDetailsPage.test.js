/* eslint-disable no-undef */
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MovieDetailsPage from '../../src/pages/MovieDetailsPage/MovieDetailsPage';

describe('Test MovieDetailsPage component', () => {
  test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<MovieDetailsPage />);
    renderer.getRenderOutput();
    expect(result.type).toBe('div');
    expect(result.props.className).toBe('movie-details');
    expect(result).toMatchSnapshot();
  })
})