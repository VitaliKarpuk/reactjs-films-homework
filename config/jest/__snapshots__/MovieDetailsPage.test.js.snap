// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MovieDetailsPage component renders correctly 1`] = `
<div
  className="movie-details"
>
  <header
    className="header"
  >
    <h1>
      FILMS
    </h1>
    <UNDEFINED />
  </header>
  <div
    className="movie-details__content"
  >
    <div
      className="movie-details__title"
    >
      <h2>
        THE JUNGLE BOOK
      </h2>
      <ul
        className="movie-details__genre"
      >
        <li>
          Adventure
        </li>
        <li>
          Drama
        </li>
        <li>
          Family
        </li>
        <li>
          Fantasy
        </li>
        <li>
          1h 46m
        </li>
      </ul>
      <Stars />
    </div>
    <div
      className="movie-details__buttons"
    >
      <ModalInfo />
      <Btn
        className="btn__watch"
        title="Watch Now"
      />
      <Btn
        className="btn__info"
        title="View Info"
      />
    </div>
  </div>
</div>
`;
